crates/uv-requirements/src/pyproject.rs

```diff
@@ -441,18 +441,18 @@ pub(crate) fn lower_requirement(
                 _ => return Err(LoweringError::MoreThanOneGitRef),
             };
 
+            // Create a PEP 508-compatible URL.
             let mut url = Url::parse(&format!("git+{git}"))?;
-            let mut given = git.to_string();
             if let Some(rev) = reference.as_str() {
                 url.set_path(&format!("{}@{}", url.path(), rev));
-                given = format!("{given}@{rev}");
             }
             if let Some(subdirectory) = &subdirectory {
                 url.set_fragment(Some(&format!("subdirectory={subdirectory}")));
-                given = format!("{given}#subdirectory={subdirectory}");
```

Comment:
@konstin - There isn't really any point in reconstructing these, I don't think? It's just gonna be the same as the serialized form of the URL. The purpose of `given` is to let us preserve environment variables (as an example) and other details that won't match the serialized representation.